installing/installing_gcp/installing-gcp-user-infra-vpc.adoc

```diff
@@ -0,0 +1,138 @@
+include::modules/installation-deployment-manager-ext-lb.adoc[leveloffset=+2]
+include::modules/installation-deployment-manager-int-lb.adoc[leveloffset=+2]
+
+include::modules/installation-creating-gcp-private-dns.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-private-dns.adoc[leveloffset=+2]
+
+include::modules/installation-creating-gcp-firewall-rules-vpc.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-firewall-rules.adoc[leveloffset=+2]
+
+include::modules/installation-creating-gcp-iam-shared-vpc.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-iam-shared-vpc.adoc[leveloffset=+2]
+
+include::modules/installation-gcp-user-infra-rhcos.adoc[leveloffset=+1]
+
+include::modules/installation-creating-gcp-bootstrap.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-bootstrap.adoc[leveloffset=+2]
+
+include::modules/installation-creating-gcp-control-plane.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-control-plane.adoc[leveloffset=+2]
+
+include::modules/installation-gcp-user-infra-wait-for-bootstrap.adoc[leveloffset=+1]
+
+include::modules/installation-creating-gcp-worker.adoc[leveloffset=+1]
+include::modules/installation-deployment-manager-worker.adoc[leveloffset=+2]
+
+include::modules/cli-installing-cli.adoc[leveloffset=+1]
+
+include::modules/cli-logging-in-kubeadmin.adoc[leveloffset=+1]
+
+include::modules/installation-approve-csrs.adoc[leveloffset=+1]
+
+include::modules/installation-gcp-user-infra-adding-ingress.adoc[leveloffset=+1]
+
+[id="installation-gcp-user-infra-vpc-adding-firewall-rules"]
+== Adding cluster-wide firewall rules
+The cluster requires several firewall rules. If you do not use a shared VPC, these rules are created by the ingress controller via the GCP cloud provider. When you use a shared VPC, you can either create cluster-wide firewall rules for all services now or create each rule based on events, when the cluster requests access. By creating each rule when the cluster requests access, you know exactly which firewall rules are required. By creating cluster-wide firewall rules, you can apply the same rule set across multiple clusters.
+
+If you chose to create each rule based on events, you must create firewall rules after you provision the cluster and during the life of the cluster when the console notifies you that rules are missing. Events that are similar to the following event are displayed, and you must add the firewall rules that are required:
+
+----
+Firewall change required by security admin: `gcloud compute firewall-rules create k8s-fw-a26e631036a3f46cba28f8df67266d55 --network example-network --description "{\"kubernetes.io/service-name\":\"openshift-ingress/router-default\", \"kubernetes.io/service-ip\":\"35.237.236.234\"}\" --allow tcp:443,tcp:80 --source-ranges 0.0.0.0/0 --target-tags exampl-fqzq7-master,exampl-fqzq7-worker --project example-project`
```

Comment:
The cluster has trouble emitting events currently. We have a bug to track it. I'm not sure if it's proper to have a problematic function described here.
Bug 1816806 - system:serviceaccount:kube-system:cloud-provider cannot create resource events

Reply:
[This comment](https://bugzilla.redhat.com/show_bug.cgi?id=1816806#c12) on the bug makes it sound like it's fixed in 4.5. If it's still an issue, I will open a separate PR to fix it.